sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Reshuffle.java

```diff
@@ -44,18 +44,18 @@
  *
  * <p>A {@link PTransform} that returns a {@link PCollection} equivalent to its input but
  * operationally provides some of the side effects of a {@link GroupByKey}, in particular
- * checkpointing, and preventing fusion of the surrounding transforms.
+ * redistribution of elements between workers, checkpointing, and preventing fusion of the
+ * surrounding transforms. Some of these side effects (e.g. checkpointing) are not portable and will
+ * not occur on workers that don't have these properties when performing a {@link GroupByKey}.
```

Comment:
s/workers/runners.

Additionally, these operational effects are not guaranteed even on runners where GroupByKey does have these side effects.